### DISCUSSION TITLE
Some debug information about 3d model

### DISCUSSION THREAD

**Discussion by noseeevil** [google/filament#3379] [Show and tell]
Hello!

Can I get some debug info about loaded model (.glb) like count of verts/tris, 
and something about graphics in the scene like a SetPass calls/batches/animations count and etc?

**Comment by cx20** [comment]
In the WebGL environment, [Spector.js](https://spector.babylonjs.com/) can be used to obtain some debugging information. Spector.js is a browser extension for Chrome and Firefox.
https://www.realtimerendering.com/blog/debugging-webgl-with-spectorjs/

To see the structure of the glTF model itself, the [glTF Tools Extension for VS Code](https://github.com/AnalyticalGraphicsInc/gltf-vscode) may be helpful.
https://github.com/AnalyticalGraphicsInc/gltf-vscode

**Comment by bejado** [comment]
You can query the number of primitives in a glTF entity with:

```
auto e = asset->getFirstEntityByName("my glTF entity");
auto& rm = engine->getRenderableManager();
size_t primitiveCount = rm.getPrimitiveCount(rm.getInstance(e));
```
